### PR TITLE
feat: Implement duplicate entity redirection logic and add correspond…

### LINKED ISSUE #2660: Complex Batch Assign - Part 2a: Duplicate redirection for spatial datasets
Following implementation of Part 1, we should add a feature that analyses if any of the new entities are duplicates of existing ones on the platform. This analysis will only be for spatial data - i.e. those with typology `geography`:

- We should use the work by @sianteesdale on duplicate analysis to dedepulicate the data - only looking at entities assigned to that provision.
- Assign the new reference to a new entity number and redirect the old entity using the old-entity CSV and status 301

### PR DESCRIPTION
**Pull request by gibahjoe**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Adds duplicate entity redirection support for spatial add-data requests. The request processor now identifies potential duplicate conservation-area entities by comparing newly transformed provision entities against existing platform entities using geometry/point matching, then includes `duplicate-candidates` in the add-data response for use in Assign Entities.

This also adds partial-ratio name similarity using `rapidfuzz`, installs Spatialite support in the processor image, and adds unit coverage for duplicate candidate generation, Datasette fetching, name similarity, and skip conditions.

## Related Tickets & Documents

- Ticket Link
- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings

Run the duplicate candidate unit tests:

```bash
cd request-processor
python3 -m pytest tests/unit/src/test_duplicates.py
```

Also verify an add-data response for a conservation-area resource includes a `duplicate-candidates` array when a new provision entity spatially matches an existing platform entity.

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests have not been included_
- [ ] I need help with writing tests

## [optional] Are there any post deployment tasks we need to perform?

None.

## [optional] Are there any dependencies on other PRs or Work?

None known.

## Summary by CodeRabbit

* **New Features**
  * Added duplicate-entity redirect suggestions to add-data processing results, including match type, references, entry/end dates, evidence, and optional name similarity.
* **Bug Fixes**
  * Duplicate-candidate analysis is fault-tolerant (logs failures and safely returns no candidates).
  * Duplicate analysis is limited to supported dataset/scope and geography specifications.
* **Chores**
  * Pinned an additional fuzzy-matching dependency and added an extra spatial database OS library to the container.
* **Tests**
  * Added unit coverage for duplicate-redirect candidate generation, matching, and key edge cases.

### CODE REVIEW

**Review by pooleycodes — APPROVED** [review]

File: request-processor/src/application/core/duplicates.py [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+    if not any(row.get(spatial_field) for row in rows):
+        return {"complete_matches": [], "single_matches": []}
+
+    fd, path = tempfile.mkstemp(suffix=".sqlite3")
+    os.close(fd)
+    conn = None
+    try:
+        import spatialite
+        from digital_land.expectations.operations.dataset import (
+            duplicate_geometry_check,
```
Comment (pooleycodes) [inline request-processor/src/application/core/duplicates.py:170]: duplicate_geometry_check is an expectations check meant to self-join on digital-land's own sqlite, here it's repurposed for a live per-request call with all the entity table creation logic that goes with it to match its internals.

Should the entity matching logic live inside digital-land-python?

File: request-processor/src/application/core/duplicates.py [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+import tempfile
+
+import requests
+from rapidfuzz import fuzz
+
+from application.logging.logger import get_logger
+
+logger = get_logger(__name__)
+
+DATASETTE_BASE_URL = "https://datasette.planning.data.gov.uk"
```
Comment (pooleycodes) [inline request-processor/src/application/core/duplicates.py]: Is it possible to fetch this from the Planning Data API instead? Also may want to add some environment matching here, i.e. development.planning.gov.uk, staging.planning etc.

File: request-processor/src/application/core/duplicates.py (hunk 1 above)
Comment (gibahjoe) [reply request-processor/src/application/core/duplicates.py:170]: I spoke with Owen, and this approach is acceptable because it supports a specific use case. If we later find that the code is needed elsewhere, we can consider moving it into Digital Land Python.

In the longer term, we should migrate to Parquet, as SQLite is less efficient for this use case and Parquet should provide better performance.

File: request-processor/src/application/core/duplicates.py (hunk 2 above)
Comment (gibahjoe) [reply request-processor/src/application/core/duplicates.py]: I’ve updated this to use environments instead.

I decided not to use the Planning Data API because retrieving all the entities would require pagination and multiple requests, which would slow down `add_data`. Since this process fetches every entity on the platform, it could also place unnecessary load on the API (if i dont paginate it) and degrade performance for its consumers.

**Review by pooleycodes — APPROVED** [review]
LGTM
